ethdb/kv/kv_snapshot.go

```diff
@@ -754,6 +770,14 @@ func (s *snCursor) Last() ([]byte, []byte, error) {
 	if err != nil {
 		return nil, nil, err
 	}
+
+	for bytes.Equal(lastDBVal, DeletedValue) {
+		lastDBKey, lastDBVal, err = s.dbCursor.Prev()
```

Comment:
must be for until !bytes.Equal(DeletedValue ?

Reply:
No, that's correct. We want to find the last non-deleted element.